### ISSUE TITLE
glitchy/blocky output

### ISSUE THREAD

**Issue by kylemcdonald** [Quasimondo/RasterFairy#13]
i'm still working with this cloud-to-grid problem, and wanted to understand rasterfairy a little better because it's so fast for larger datasets. but i'm having trouble getting smooth output. i generated some toy data:

```python
import numpy as np
side = 85
n = side * side
np.random.seed(0)
lhs = np.random.uniform(low=-1, high=+1, size=(n, 2))
lhs = np.cumsum(lhs, axis=0)
lhs -= lhs.min(axis=0)
lhs /= lhs.max(axis=0)
lhs *= (side - 1)
```

the output from rasterfairy looks like this:

![download-4](https://user-images.githubusercontent.com/157106/50406702-4e278f80-077e-11e9-8d38-1b1db3e3477a.png)

while a linear assignment solver shows the optimum is:

![download-2](https://user-images.githubusercontent.com/157106/50406705-58e22480-077e-11e9-84b6-6a41003c3634.png)

is there any way to get a better result out of rasterfairy, or is this "blocky" solution part of the algorithm? with some linear assignment algorithms there are "approximation" knobs that let you tune the run time vs solution accuracy, and i was wondering if rasterfairy has something similar i could tweak.

**Comment by Quasimondo** [comment]
Yeah the blockiness is the result of the way Rasterfairy subdivides the cloud which is purely based on vertical and horizontal splits. This is fast but can lead to sub-optimal results depending on the shape of the cloud. Ideally the subdivision algorithm should be replaced with something more intelligent that takes the shape of the cloud into account, but at the moment I can only offer the warp routines that allow to pre-process the cloud.

I have added another demo notebook that shows how to use the coonswarp class:
https://github.com/Quasimondo/RasterFairy/blob/master/examples/Raster%20Fairy%20Demo%202.ipynb

Not sure though if the results are any better, maybe you can compare it with your code
